src/forwarders.js

```diff
@@ -387,33 +393,46 @@ export default function Forwarders(mpInstance, kitBlocker) {
         }
     };
 
-    this.callSetUserAttributeOnForwarders = function(key, value) {
-        if (kitBlocker && kitBlocker.isAttributeKeyBlocked(key)) {
+    this.handleForwarderUserAttributes = function(key, value, functionNameKey) {
+        if (
+            (kitBlocker && kitBlocker.isAttributeKeyBlocked(key)) ||
+            !mpInstance._Store.activeForwarders.length
+        ) {
             return;
         }
 
-        if (mpInstance._Store.activeForwarders.length) {
-            mpInstance._Store.activeForwarders.forEach(function(forwarder) {
-                if (
-                    forwarder.setUserAttribute &&
-                    forwarder.userAttributeFilters &&
-                    !mpInstance._Helpers.inArray(
-                        forwarder.userAttributeFilters,
-                        mpInstance._Helpers.generateHash(key)
-                    )
-                ) {
-                    try {
-                        var result = forwarder.setUserAttribute(key, value);
+        mpInstance._Store.activeForwarders.forEach(function(forwarder) {
+            const forwarderFunction = forwarder[functionNameKey];
+            if (
+                forwarderFunction &&
```

Comment:
I think you can break out of this nested if by just returning after checking for the inverse:

```
if (
  !forwarderFunction &&
  !mpInstance._Helpers.isFilteredUserAttribute(
    key,
    forwarder.userAttributeFilters
  )
) {
  return;
}

Reply:
@alexs-mparticle  also, I changed the operator here from `&&` to `||` since if it fails in either case we should continue with the next forwarder, correct me if I'm wrong.